actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -975,4 +986,64 @@ private void setDefaultUserProfileVisibility(Map<String, Object> userMap) {
         JsonKey.DEFAULT_PROFILE_FIELD_VISIBILITY,
         ProjectUtil.getConfigValue(JsonKey.SUNBIRD_USER_PROFILE_FIELD_DEFAULT_VISIBILITY));
   }
+
+  @SuppressWarnings("unchecked")
+  public static String getFrameworkId(String channel) {
+    String frameworkId = DataCacheHandler.getchannelFrameworkMap().get(channel);
+    if (frameworkId != null) {
+      Map<String, Object> resultMap =
+          ContentStoreUtil.getReadDetails(channel, JsonKey.SUNBIRD_CHANNEL_READ_API);
+      Map<String, Object> results = (Map<String, Object>) resultMap.get(JsonKey.RESULT);
+      Map<String, Object> channelDetails = (Map<String, Object>) results.get(JsonKey.CHANNEL);
+      frameworkId = (String) channelDetails.get(JsonKey.DEFAULT_FRAMEWORK);
+      DataCacheHandler.updateChannelFrameworkMap(channel, frameworkId);
+      return frameworkId;
+    } else {
+      throw new ProjectCommonException(
+          ResponseCode.errorNoFrameworkFoundForUserChannel.getErrorCode(),
+          ResponseCode.errorNoFrameworkFoundForUserChannel.getErrorMessage(),
+          ResponseCode.RESOURCE_NOT_FOUND.getResponseCode());
+    }
+  }
+
+  public static Map<String, List<Map<String, String>>> getFrameworkDetails(String frameworkId) {
+    if (DataCacheHandler.getFrameworkMap().get(frameworkId) == null) {
+      getFrameworkDataAndCache(frameworkId);
+    }
+    return DataCacheHandler.getFrameworkMap().get(frameworkId);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static void getFrameworkDataAndCache(String frameworkId) {
+    Map<String, Object> response =
+        ContentStoreUtil.getReadDetails(frameworkId, JsonKey.SUNBIRD_FRAMEWORK_READ_API);
+    Map<String, List<Map<String, String>>> frameworkCacheMap = new HashMap<>();
+    List<String> supportedfFields = DataCacheHandler.getFrameworkFieldsConfig().get(JsonKey.FIELDS);
+    Map<String, Object> result = (Map<String, Object>) response.get(JsonKey.RESULT);
+    Map<String, Object> frameworkDetails = (Map<String, Object>) result.get(JsonKey.FRAMEWORK);
+    List<Map<String, Object>> frameworkCategories =
+        (List<Map<String, Object>>) frameworkDetails.get(JsonKey.CATEGORIES);
+    for (Map<String, Object> frameworkCategoriesValue : frameworkCategories) {
+      String frameworkField = (String) frameworkCategoriesValue.get(JsonKey.CODE);
+      List<Map<String, String>> listOfFields = new ArrayList<>();
+      if (supportedfFields.contains(frameworkField)
+          || frameworkField.equalsIgnoreCase(JsonKey.GRADE_LEVEL)) {
```

Comment:
Don't have any special check for GRADE_LEVEL. Update configuration instead.

e.g. userProfileConfig.mandatoryFields = ["board", "gradeLevel", "medium"]